[DirectX][Docs] Add DXILIntrinsicExpansion  Pass to DXILArchitecture.rst (#86198)
llvm/llvm-project: llvm/docs/DirectX/DXILArchitecture.rst

logic with both the DirectX backend and with Clang's codegen of HLSL
support as much as possible.

The DirectX Intrinsic Expansion Pass
====================================
There are intrinsics that don't map directly to DXIL Ops. In some cases
an intrinsic needs to be expanded to a set of LLVM IR instructions. In
other cases an intrinsic needs modifications to the arguments or return

> values of a DXIL Op. This pass (`DXILIntrinsicExpansion.cpp`) handles all 

Reviewer: ```suggestion
values of a DXIL Op. The `DXILIntrinsicExpansion` pass handles all 
```

I think this is more consistent with the other pass descriptions above.

> the cases where our intrinsics don't have a one to one mapping. You should 
> also consider using this pass when the expansion is specific to DXIL so 
> as to keep implementation details out of CodeGen. Finally, there is an 
> expectation that we maintain vectors through this pass. Therefore, best 

Reviewer: ```suggestion
the cases where our intrinsics don't have a one to one mapping. This pass 
may also be used when the expansion is specific to DXIL so 
as to keep implementation details out of CodeGen. Finally, there is an 
expectation that we maintain vectors through this pass. Therefore, best 
```

Suggestion is to try and keep similar voice to rest of doc.

> expectation that we maintain vectors through this pass. Therefore, best 

Reviewer: > vectors

Is "vectors" in this context unambiguous enough for the intended audience?  Would something like "vector types" be clearer?
